BrickController2/BrickController2/DeviceManagement/DeviceManager.cs

```diff
@@ -54,10 +54,16 @@ public async Task LoadDevicesAsync()
                 var deviceDTOs = await _deviceRepository.GetDevicesAsync();
                 foreach (var deviceDTO in deviceDTOs)
                 {
-                    var device = _deviceFactory(deviceDTO.DeviceType, deviceDTO.Name, deviceDTO.Address, deviceDTO.DeviceData);
-                    if (device != null)
+                    try
+                    {
+                        var device = _deviceFactory(deviceDTO.DeviceType, deviceDTO.Name, deviceDTO.Address, deviceDTO.DeviceData);
```

Comment:
I guess this could be better solved in the factory itself, may be `ResolveOptionalKeyed`

https://github.com/vicocz/brickcontroller2/blob/81e132834fe93c7d0bbd3dc010f8c1373c1a1d13/BrickController2/BrickController2/DeviceManagement/DI/DeviceManagementModule.cs#L30